lib/trino-hdfs/src/main/java/io/trino/hdfs/s3/TrinoS3FileSystem.java

```diff
@@ -1148,6 +1150,10 @@ private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configurati
             return new AWSStaticCredentialsProvider(credentials.get());
         }
 
+        if (conf.getBoolean(S3_USE_WEB_IDENTITY_TOKEN_CREDENTIALS_PROVIDER, false)) {
```

Comment:
this reads from Hadoop config, not Trino's FS config. How is this exposed to users?

Reply:
It is exposed through the Hadoop config. See my previous comment on how I tested it.